src/contracts/core/StakeRootCompendium.sol

```diff
@@ -234,226 +202,155 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
+        require(
+            numStrategiesBefore + strategiesAndMultipliers.length <= MAX_NUM_STRATEGIES, 
+            "StakeRootCompendium.setStrategiesAndMultipliers: too many strategies"
+        );
+        // set the strategies and multipliers for the operator set
+        for (uint256 i = 0; i < strategiesAndMultipliers.length; i++) {
+            operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].set(
+                address(strategiesAndMultipliers[i].strategy), uint256(strategiesAndMultipliers[i].multiplier)
             );
         }
-
-        _updateTotals(numStrategiesBefore, operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length());
+        _updateTotalStrategies(numStrategiesBefore, operatorSet);
     }
 
-    function _removeOperatorSet(IAVSDirectory.OperatorSet memory operatorSet) internal {
-        if(operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length() != 0) {
-            bytes32[] memory strategyBytes = operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId]._inner._keys.values();
-            IStrategy[] memory strategies = new IStrategy[](strategyBytes.length);
-            // todo: better way to do this?
-            for (uint256 i = 0; i < strategyBytes.length; i++) {
-                strategies[i] = IStrategy(address(uint160(uint256(strategyBytes[i]))));
-            }
-
-            // remove the strategies and multipliers for the operator set
-            _removeStrategiesAndMultipliers(
-                operatorSet,
-                strategies
-            );
-        }
-
-        IAVSDirectory.OperatorSet memory substituteOperatorSet = operatorSets[operatorSets.length - 1];
-        uint224 operatorSetIndex = operatorSetToIndex[operatorSet.avs][operatorSet.operatorSetId].latest();
-        operatorSets[operatorSetIndex] = substituteOperatorSet;
-        operatorSets.pop();
+    function _updateTotalStrategies(uint256 numStrategiesBefore, IAVSDirectory.OperatorSet memory operatorSet) internal {
+        totalStrategies = totalStrategies - numStrategiesBefore + operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length();
+            totalChargeSnapshot.push(
+            uint32(block.number), 
+            uint224(totalStrategies * chargePerProofLinear + chargePerProofConstant)
+        );
+    }
 
-        // update the index of the operator sets
-        operatorSetToIndex[operatorSet.avs][operatorSet.operatorSetId].push(uint32(block.timestamp), REMOVED_INDEX);
-        operatorSetToIndex[operatorSet.avs][operatorSet.operatorSetId].push(uint32(block.timestamp), operatorSetIndex);
+    function _setChargePerProof(uint64 _chargePerProofLinear, uint64 _chargePerProofConstant) internal {
+        _updateCumulativeCharges();
+        chargePerProofLinear = _chargePerProofLinear;
```

Comment:
we need to update the total Charge here

Reply:
yeah, i overlooked some total charging related logic, will put it in there